### PR TITLE
Adding fybrik-python-vault library.

### LINKED ISSUE #1444: Add python library for vault
Adding a python library to Fybrik repo to help module's writers to fetch secrets and credentials from vault.

### PR DESCRIPTION
**Pull request by Mohammad-nassar10**
Adding a python library to Fybrik repo to help module's writers fetch secrets and credentials from vault. Also, Publishing the python package to `PyPi` using github actions. 

Closes #1444 

Signed-off-by: mohammad-nassar10 <mohammad.nassar@ibm.com>

### CODE REVIEW

File: python/vault/fybrik_python_vault/__init__.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+#
+# Copyright 2022 IBM Corp.
+# SPDX-License-Identifier: Apache-2.0
+#
+import requests
+from fybrik_python_logging import logger, Error, DataSetID, ForUser
+
+def test_func():
```
Comment (cdoron) [inline python/vault/fybrik_python_vault/__init__.py:8]: Seems like test_func() doesn't do anything, right?

File: python/vault/fybrik_python_vault/__init__.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        else:
+            logger.error("Malformed secret response. Expected the 'data' field in JSON",
+                         extra={DataSetID: datasetID, ForUser: True})
+    else:
+        logger.error("Error reading credentials from vault",
+            extra={Error: str(response.status_code) + ': ' + str(response.json()),
+                   DataSetID: datasetID, ForUser: True})
+    return None
+
+def get_credentials_from_vault(vault_credentials, datasetID):
```
Comment (cdoron) [inline python/vault/fybrik_python_vault/__init__.py:68]: I think that get_credentials_from_vault() should not be part of the library. What do you think?

File: python/vault/fybrik_python_vault/__init__.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    jwt_file_path = vault_credentials.get('jwt_file_path', '/var/run/secrets/kubernetes.io/serviceaccount/token')
+    jwt = get_jwt_from_file(jwt_file_path)
+    vault_address = vault_credentials.get('address', 'https://localhost:8200')
+    secret_path = vault_credentials.get('secretPath', '/v1/secret/data/cred')
+    vault_auth = vault_credentials.get('authPath', '/v1/auth/kubernetes/login')
+    role = vault_credentials.get('role', 'demo')
+    logger.trace('getting vault credentials',
```
Comment (cdoron) [inline python/vault/fybrik_python_vault/__init__.py:75]: I suggest moving this log message up to get_raw_secret_from_vault()

File: python/vault/fybrik_python_vault/__init__.py (hunk 1 above)
Comment (Mohammad-nassar10) [reply python/vault/fybrik_python_vault/__init__.py]: right, I removed it.

**Comment by shlomitk1** [comment]
Please link to the relevant issue.

**Review by cdoron — APPROVED** [review]

**Review by cdoron — APPROVED** [review]

**Pull request merged**